controllers/volume/volume_controller.go

```diff
@@ -553,6 +558,31 @@ func (r *Reconciler) processAttachments(
 			continue
 		}
 
+		// If VM hardware version doesn't meet minimal requirement, don't create CNS attachment.
+		// We only fetch the hardware version when this is the first time to create the CNS attachment.
+		//
+		// This change is along with removing VMI hardware version check from VM validation webhook.
+		// We used to deny requests if a PVC is specified in the VM spec while VMI hardware version is not supported.
+		// In this case, no attachments can be created if VM hardware version doesn't meet the requirement.
+		// So, at this point, we can safely assume that all VMs with attachments created have passed the
+		// hardware version check.
```

Comment:
Consider placing the highlighted comment after this `if` statement for better clarity.